src/dataflow/src/render/reduce.rs

```diff
@@ -1063,6 +1066,135 @@ where
         })
 }
 
+/// Accumulates values for the various types of accumulable aggregations.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+enum Accumulator {
+    /// Accumulates boolean values.
+    Bool {
+        /// The number of `true` values observed.
+        trues: isize,
+        /// The number of `false` values observed.
+        falses: isize,
+        /// The total number of values observed.
+        total: isize,
+    },
+    /// Accumulates the count of non-NULL values.
+    Count {
+        /// The number of non-NULL values observed.
+        non_nulls: isize,
+        /// The total number of values observed.
+        ///
+        /// This value is not actually observed directly by the code in this
+        /// module, but it serves to ensure that `is_zero()` returns false when
+        /// at least one non-NULL value has been observed. This is required to
```

Comment:
should this be "when at least one value ..."? It seems like it is here to ensure that *null* values get noticed, rather than non-null values?

Reply:
Oops, yes. This text is no longer present in the latest revision though.